phoebe/atmospheres/passbands.py

```diff
@@ -1721,50 +1698,116 @@ def Imu(self, query_pts, atm='ck2004', ldatm='ck2004', ldint=None, ld_func='inte
+                raise ValueError(f'{atm=} and {ld_func=} are incompatible with {ldatm=}.')
+
+            elif atm in ['ck2004', 'phoenix', 'tmap_sdO', 'tmap_DA', 'tmap_DAO', 'tmap_DO']:
+                if f'{atm}:Imu' not in self.content:
+                    raise ValueError(f'{atm=} tables are not available in the {self.pbset}:{self.pbname} passband.')
+
+                ndpolants = self.ndp[atm].ndpolate(f'imu@{intens_weighting}', query_pts, extrapolation_method=atm_extrapolation_method)
+                log10imus_atm = ndpolants['interps']
+                dists = ndpolants.get('dists', np.zeros_like(log10imus_atm))
+
+                if np.any(dists > dist_threshold) and blending_method == 'blackbody':
+                    off_grid = (dists > dist_threshold).flatten()
+                    # print(f'{query_pts.shape=} {off_grid.shape=}')
+
+                    ints_bb = self.Imu(
+                        query_pts=query_pts[off_grid],
+                        atm='blackbody',
+                        ldatm=ldatm,
+                        ldint=ldint,
+                        ld_func=ld_func,
+                        ld_coeffs=ld_coeffs,
+                        intens_weighting=intens_weighting,
+                        atm_extrapolation_method=atm_extrapolation_method,
+                        ld_extrapolation_method=ld_extrapolation_method
+                    )
+                    log10imus_bb = np.log10(ints_bb['inorms'])
 
-            if atm not in self.content and f'{atm}:Imu' not in self.content:
-                raise ValueError(f'atm={atm} tables are not available in the {self.pbset}:{self.pbname} passband.')
+                    log10imus_blended = log10imus_atm.copy()
+                    log10imus_blended[off_grid] = (np.minimum(dists[off_grid], blending_margin) * log10imus_bb + np.maximum(blending_margin-dists[off_grid], 0) * log10imus_atm[off_grid])/blending_margin
+
+                    intensities = 10**log10imus_blended
+                else:
+                    intensities = 10**log10imus_atm
+
+                return intensities
 
-            if return_nanmask:
-                intensities, nanmask = self._log10_Imu(atm=atm, query_pts=query_pts, intens_weighting=intens_weighting, atm_extrapolation_method=atm_extrapolation_method, ld_extrapolation_method=ld_extrapolation_method, blending_method=blending_method, raise_on_nans=raise_on_nans, return_nanmask=return_nanmask)
-                intensities = (10**intensities, nanmask)
             else:
-                intensities = 10**self._log10_Imu(atm=atm, query_pts=query_pts, intens_weighting=intens_weighting, atm_extrapolation_method=atm_extrapolation_method, ld_extrapolation_method=ld_extrapolation_method, blending_method=blending_method, raise_on_nans=raise_on_nans, return_nanmask=return_nanmask)
+                # anything else we need to special-handle for ld_func == 'interp'?
+                pass
```

Comment:
this task was meant to be offloaded to a test suite (in the making).

Reply:
what would get to this else?  An unsupported value for `atm`?

Reply:
any combination of atm and ldatm that isn't explicitly handled, including invalid atm/ldatm values. Instead of pass we could raise an error but I figured I'd write a test case that goes through a cartesian product of all atmospheres for atm and ldatm, including non-existing ones, and see if anything reaches this block.

Reply:
ok, can you please make sure there is a ticket for those test cases so they aren't forgotten?